portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -13,20 +13,37 @@
 // limitations under the License.
 
 package com.google.sps.servlets;
+import com.google.gson.Gson;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/greeting")
+@WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> messages;
+  private int curIndex;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<String>();
```

Comment:
Another alternative of initializing the list is by doing inline initialization.
What do you think the difference between the two methods?
```java
private List<String> messages = Arrays.asList("Example message 1", .....);
```

Reply:
My understanding is that using `Arrays.asList()` requires you to initialize an array to pass into the method. While I know that the underlying mechanism for ArrayList uses arrays, I do not know enough about the implementation for the `Arrays.asList` method. If the array passed in is directly used as the container for the ArrayList, I can see this method being faster. On the other hand, if the underlying implementation of `Arrays.asList` just loops through the array and adds the element one by one (which is what happens if you append two `LinkedLists`), this method will be slower because now you need to initialize an extra object.

Reply:
Thanks for your explanation Daekoon.
I was referring more towards initializing it during servlet init lifecycle vs class instantiation level.
The detail of `Arrays.asList` is not what I'd like to point out.

Basically, I would like you to think about the trade-off of placing certain piece of code at different places cause it may result in differences during runtime.